docs/style.md

```diff
@@ -539,6 +539,245 @@ import { unstable_Pagination as Pagination } from 'carbon-components-react';
+- Accessibility
+- End to end tests
+- Server side rendering
+
+Each of these are separated into individual files. In some cases the syntax may
+be slightly different and separate files make this easier to understand.
+Additionally separate file types can be more easily globbed to only run certain
+tests in certain environments (local, CI, Pre-release checks, etc).
+
+| File name                    | Category                        |
+| ---------------------------- | ------------------------------- |
+| `ComponentName-test.js`      | General component functionality |
+| `ComponentName-test.a11y.js` | Accessibility testing           |
+| `ComponentName-test.e2e.js`  | End to end tests                |
+| `ComponentName-test.ssr.js`  | Server side rendering           |
+
+There are corresponding commands to run all categories, individual categories,
+or a combination. Depending on your shell, modifiers can be used to run two
+commands one after another. Refer to the documentation of your shell.
+
+| Command                           | Corresponding test category                                          |
+| --------------------------------- | -------------------------------------------------------------------- |
+| `yarn test`                       | All categories                                                       |
+| `yarn test:unit`                  | Only component unit tests                                            |
+| `yarn test:a11y`                  | Only accessibility tests                                             |
+| `yarn test:e2e`                   | Only end to end tests                                                |
+| `yarn test:ssr`                   | Only server side tests                                               |
+| `yarn test:a11y && yarn test:e2e` | In `bash` via `&&`: Run the a11y tests, and if they succeed, run e2e |
+
+#### Recipes
+
+Below are some common receipes for component testing. Many of the pattern/syntax
+details contained within these recipes are enforced via eslint rules declared in
+`eslint-config-carbon`.
+
+##### `ComponentName-test.js`
+
+- Use `@testing-library/react`
+  - Render using `render()`
+  - Query using `screen()`, prefer
+    [queries accessible to everyone](https://testing-library.com/docs/queries/about#priority)
+  - Simulate events with `userEvent`
+- Format with `describe`/`it` blocks
+- Use [`jest-dom`](https://github.com/testing-library/jest-dom) matchers for
+  assertions
+- Keyboard navigation/interaction event testing is not well supported through
+  JSDOM. Testing this in a simulated browser environment, such as Cypress (e2e),
+  is a better environment for this type of testing.
+
+```js
+import { render, screen, findByLabel } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { ComponentName } from '../ComponentName';
+
+describe('ComponentName', () => {
+  it('should render with default props', () => {
```

Comment:
I'm not sure how useful these kinds of tests are, do you think we need them with some of the other tests that we're writing?

Reply:
Probably not. Aim is to get some sort of absolute baseline for components, like what is the ground level? I think all components should have a className placed on the root node, maybe that's where all component files should start?

Reply:
@tay1orjones nice, I like it. Especially for this file that seems to be testing the "component API" if we're using Cypress for behavior